### PR TITLE
Update libthemis formula to 0.15.1 

### PR DESCRIPTION
**Pull request by radetsky**
1. Updated formula code to the latest version of Themis
2. Replaced openssl 1.1 to 3 
3. Tested with C, Python 
4. Fix a few formula warnings after the `brew audit`

### CODE REVIEW

File: Formula/libthemis.rb [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 
   def install
-    ENV['ENGINE'] = 'openssl'
-    ENV['ENGINE_INCLUDE_PATH'] = Formula['openssl@1.1'].include
-    ENV['ENGINE_LIB_PATH'] = Formula['openssl@1.1'].lib
-    ENV['PREFIX'] = prefix
-    system 'make', 'install'
-    if build.with? 'cpp'
-      system 'make', 'themispp_install'
+    ENV["ENGINE"] = "openssl"
```
Comment (vixentael) [inline Formula/libthemis.rb:16]: any particular reason why ruby prefers `"` to `'`? 🤔

**Review by vixentael — APPROVED** [review]
let's wait for @shadinua review

File: Formula/libthemis.rb [hunk 2]
```diff
@@ -1,34 +1,34 @@
 class Libthemis < Formula
-  desc 'High-level cryptographic primitives'
-  homepage 'https://www.cossacklabs.com/themis'
-  head 'https://github.com/cossacklabs/themis.git'
-  url 'https://github.com/cossacklabs/themis/archive/0.14.0.tar.gz'
-  sha256 '2efb793e0ef604fb97258b07671a83135ad9229d83b92d7758b43510dcc6cb07'
-  version '0.14.0'
+  desc "High-level cryptographic primitives"
+  homepage "https://www.cossacklabs.com/themis"
+  url "https://github.com/cossacklabs/themis/archive/0.15.1.tar.gz"
```
Comment (vixentael) [inline Formula/libthemis.rb:4]: we will need to upload gz to github, right?

File: Formula/libthemis.rb (hunk 1 above)
Comment (shadinua) [reply Formula/libthemis.rb:16]: The same question. There are many discussions above the question, but in Ruby we use `"` when we expecting string interpolation, and `'` when we don't. So, using `"` in all places brings potential risks to get interpolation where we don't expect it. I wouldn't suggest such approach.

**Review by shadinua — COMMENTED** [review]
All other looks expected to me, thank you.

File: Formula/libthemis.rb (hunk 1 above)
Comment (vixentael) [reply Formula/libthemis.rb:16]: @radetsky let's switch back to the `'`?

Comment (radetsky) [reply Formula/libthemis.rb:16]: Ok.

File: Formula/libthemis.rb (hunk 2 above)
Comment (radetsky) [reply Formula/libthemis.rb:4]: No. It is a GitHub feature. It is already there.

**Comment by radetsky** [comment]
It is a little bit clear now. Thank you for the review.

**Review by vixentael — APPROVED** [review]

**Review by shadinua — APPROVED** [review]
Great, thank you!

**Pull request merged**